### PR TITLE
Feature/k2 client

### PR DESCRIPTION
**Pull request by stepelle**
Minor modifications to the startBody.cpp file. The package now uses catkin build.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,39 +8,80 @@ include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
...
 #  RelWithDebInfo : w/ debug symbols, w/ optimization
 #  MinSizeRel     : w/o debug symbols, w/ optimization, stripped binaries
-#set(ROS_BUILD_TYPE RelWithDebInfo)
 
-rosbuild_init()
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake/Modules/)
 
 find_package(Jsoncpp REQUIRED)
+find_package(catkin REQUIRED COMPONENTS
+  rospy
```
Comment (psigen) [inline CMakeLists.txt]: Change this to 4 space indent.

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,39 +8,80 @@ include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
...
+
+catkin_package(
+    INCLUDE_DIRS include
+    #LIBRARIES
+    #    ${PROJECT_NAME}
+    CATKIN_DEPENDS
+        geometry_msgs
+        std_msgs
+    DEPENDS
+        Jsoncpp
```
Comment (psigen) [inline CMakeLists.txt]: @mkoval: is this key already covered by the ROS or PR `rosdep.yaml`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -7,39 +8,80 @@ include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
...
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 #set the default path for built libraries to the "lib" directory
 set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
-#uncomment if you have defined messages
-rosbuild_genmsg()
-#uncomment if you have defined services
-#rosbuild_gensrv()
-
 set(includeFolders ${PROJECT_SOURCE_DIR}/include/k2_client)
```
Comment (psigen) [inline CMakeLists.txt]: Can we do
`set(${PROJECT_NAME}_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/include/k2_client)`?

File: CMakeLists.txt [hunk 4]
```diff
@@ -7,39 +8,80 @@ include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
...
-
 set(RGBsourceFiles ${PROJECT_SOURCE_DIR}/src/startRGB.cpp)
```
Comment (psigen) [inline CMakeLists.txt]: If we only have one source file for each of these, perhaps we can just declare them inline in the `add_executable` block.

**Comment by psigen** [comment]
Thanks Stefania!  Can you make a few changes to the `CMakeLists.txt` formatting so I can merge?

File: src/startBody.cpp [hunk 5]
```diff
@@ -43,18 +45,18 @@ int main(int argC,char **argV)
...
 	while(ros::ok())
 	{
-		mySocket.readData();
+		mySocket.readData();        
 		char *jsonCharArrayPtr;
 		char *socketCharArrayPtr;
 		jsonCharArrayPtr = jsonCharArray;
 		socketCharArrayPtr = mySocket.mBuffer;
-		iconv(charConverter,&socketCharArrayPtr,&readSkipSize,&jsonCharArrayPtr,&stringSize);
+        iconv(charConverter,&socketCharArrayPtr,&readSkipSize,&jsonCharArrayPtr,&stringSize);
```
Comment (psigen) [inline src/startBody.cpp]: I think tabs and spaces got mixed in this file.  Can we convert to spaces-only please?

**Comment by psigen** [comment]
This PR will consolidate and supersede the bugfixes from:
https://github.com/personalrobotics/k2_client/pull/4
https://github.com/personalrobotics/k2_client/pull/5

File: CMakeLists.txt (hunk 1 above)
Comment (stepelle) [inline CMakeLists.txt]: Fixed

File: CMakeLists.txt (hunk 3 above)
Comment (stepelle) [inline CMakeLists.txt]: Fixed eliminating set line.

File: CMakeLists.txt (hunk 4 above)
Comment (stepelle) [inline CMakeLists.txt]: Fixed.

File: src/startBody.cpp (hunk 5 above)
Comment (stepelle) [inline src/startBody.cpp]: Fixed.

**Comment by stepelle** [comment]
Pras, I made the changes.

**Pull request merged**